cursive-core/src/views/dialog.rs

```diff
@@ -745,86 +855,186 @@ impl Dialog {
 
 impl View for Dialog {
     fn draw(&self, printer: &Printer) {
-        // This will be the buttons_height used by the buttons.
-        let buttons_height = match self.draw_buttons(printer) {
-            Some(height) => height,
-            None => return,
-        };
+        match self.vertical_button_orientation {
```

Comment:
Same here, we should be able to abstract over the orientation and have a single implementation.

Reply:
I remade the draw implementation. It has a single branch now, with some additional steps if the orientation is vertical.